### DISCUSSION TITLE
Getting the objective value of a specific solution.

### DISCUSSION THREAD

**Discussion by thomasgudjonwright** [ANL-CEEESA/UnitCommitment.jl#20] [Help & FAQ]
Hi! 

I apologize if I am missing something obvious, but I am wondering how one would access the objective value of a given solution. I see that you have a `validate` method which verifies the candidate solution is feasible. I first thought I could do this via `JuMP.value`, however variable names seem to all be `noname` in `model[:obj]` so I am unable to create a function from the key to values in the solution Dict. Any thoughts or easy solutions I am missing?

Thanks in advance!

Tom

**Comment by iSoron** [comment]
Hi @thomasgudjonwright, after you have built the model with `UnitCommitment.build_model(...)` and solved it with `UnitCommitment.optimize!(model)`, you can query the objective value by using `JuMP.objective_value(model)`. For example:

```julia
using Cbc
using JuMP
using UnitCommitment

instance = UnitCommitment.read_benchmark("matpower/case118/2017-02-01")
model = UnitCommitment.build_model(
    instance=instance,
    optimizer=Cbc.Optimizer,
)
UnitCommitment.optimize!(model)
@show JuMP.objective_value(model)
```

If you are trying to query the value of each decision variable instead, that's [described in our documentation](https://anl-ceeesa.github.io/UnitCommitment.jl/0.2/model/#accessing-decision-variables).

**Reply by thomasgudjonwright** [reply]
Thanks for the quick reply, @iSoron ! I may not have been clear, sorry! I meant if I have used some external model to generate a solution. ie the solution is not stored within the optimized model, but is being loaded from a `.json` for example. Does this explanation help clarify my question?
